test-d/keys-of-union.ts

```diff
@@ -35,3 +35,25 @@ type Expected2 = 'common' | 'a' | 'b' | 'c';
 declare const actual2: KeysOfUnion<Example2>;
 
 expectType<Expected2>(actual2);
+
+// With property modifiers
+declare const actual3: KeysOfUnion<{a?: string; readonly b: number} | {a: number; b: string}>;
+expectType<'a' | 'b'>(actual3);
+
+// KeysOfUnion<T> should NOT be assignable to keyof T
+type Assignability1<T, _K extends keyof T> = unknown;
```

Comment:
add test:
```ts
type Assignability1<T extends UnknownRecord, K extends keyof T> = unknown;
type Test1<T extends UnknownRecord> = Assignability1<T, KeysOfUnion<T>>; // should error
```

Reply:
@Emiyaaaaa The test you suggested doesn’t seem to produce an error:

```ts
type Assignability5<T extends UnknownRecord, _K extends keyof T> = unknown;
type Test5<T extends UnknownRecord> = Assignability5<T, KeysOfUnion<T>>; // doesn't error
```

<br>

However, adding slightly different constraints does produce the expected error:
```ts
type Assignability5<T extends Record<string, unknown>, _K extends keyof T> = unknown;
// @ts-expect-error
type Test5<T extends Record<string, unknown>> = Assignability5<T, KeysOfUnion<T>>; // Does error
```

```ts
type Assignability5<T extends object, _K extends keyof T> = unknown;
// @ts-expect-error
type Test5<T extends object> = Assignability5<T, KeysOfUnion<T>>; // Does error
```

Reply:
I don't have idea right now, maybe it's a bug of Typescript I'm not sure. I'll try to look into this interesting issue later